### PR TITLE
Adds query for user

### PR DESCRIPTION
**Pull request by gemcfadyen**
@Maikon : This pull request adds the functionality for querying a particular user.
If you have any feedback please comment on the PR and I will work it in.

Outstanding:
- Removing duplication between TransformUserResponse and TransformRepoResponse
- I've not tried deploying it yet, will do this in the morning.

NB: All out of hits on the real github api for today

### CODE REVIEW

File: web/github/transform_user_response.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  def convert(raw_response) do
+    Poison.Parser.parse!(raw_response)
+    |> downcase_user
+    |> convert_keys_to_atoms
+    |> remove_updated_at_key
+    |> convert_map_to_struct
+  end
+
+  defp downcase_user(user) do
```
Comment (gemcfadyen) [inline web/github/transform_user_response.ex]: Most of the private functions are the same as in `TransformRepoResponse` so am thinking of extracting them out into some sort of response transformer that both can use.

File: web/github/fetch_user.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  @client Application.get_env(:github_service, :client)
+
+  def with_username(name) do
+    user = Storage.find_user(name)
+
+    transform_user(user, found_user?(user))
+  end
+
+  defp transform_user(user, true), do: user
```
Comment (Maikon) [inline web/github/fetch_user.ex]: We can pattern-match on the return value of `Storage.find_user(name)` directly here instead and that would remove the `found_user?` function.

`user -> user` , `nil -> fetch_from_api`

File: test/controllers/github_controller_test.exs [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+
+    assert response.status == 200
+  end
+
+  test "user endpoint responds with user data in json" do
+    conn = conn(:get, "/users/hackeryou")
+
+    response = GithubService.Router.call(conn, [])
+
+    stored_repos = Storage.find_user("hackeryou")
```
Comment (Maikon) [inline test/controllers/github_controller_test.exs]: `found_user`?

File: web/github/transform_user_response.ex (hunk 1 above)
Comment (Maikon) [inline web/github/transform_user_response.ex]: Yep 👍

File: web/github/user.ex [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  use GithubService.Web, :model
+  @derive {Poison.Encoder, only: [:login, :public_repos, :repos_url]}
+
+  schema "users" do
+    field :login, :string
+    field :public_repos, :integer
+    field :repos_url, :string
+
+    timestamps
+
```
Comment (Maikon) [inline web/github/user.ex]: Empty line.

**Comment by gemcfadyen** [comment]
@Maikon I've addressed the feedback from the PR.

I've also added a README. Please can you take a look and let me know if I've missed anything.

I'll merge these changes this afternoon. Thanks!

**Pull request merged**